tests/semantic_analyser.cpp

```diff
@@ -4507,34 +4507,6 @@ kprobe:f { percpu_kaddr("nonsense"); }
 )" });
 }
 
-TEST_F(SemanticAnalyserBTFTest, call_socket_cookie)
-{
-  test("fentry:tcp_shutdown { $ret = socket_cookie(args.sk); }");
-  test("fexit:tcp_shutdown { $ret = socket_cookie(args.sk); }");
-
-  test("fentry:tcp_shutdown { $ret = socket_cookie(); }", Error{ R"(
-stdin:1:30-45: ERROR: socket_cookie() requires one argument (0 provided)
-fentry:tcp_shutdown { $ret = socket_cookie(); }
-                             ~~~~~~~~~~~~~~~
-)" });
-  test("fentry:tcp_shutdown { $ret = socket_cookie(args.how); }", Error{ R"(
-stdin:1:30-53: ERROR: socket_cookie() only supports 'struct sock *' as the argument (int provided)
-fentry:tcp_shutdown { $ret = socket_cookie(args.how); }
-                             ~~~~~~~~~~~~~~~~~~~~~~~
-)" });
-  test("fentry:func_1 { $ret = socket_cookie(args.foo1); }", Error{ R"(
-stdin:1:24-48: ERROR: socket_cookie() only supports 'struct sock *' as the argument ('struct Foo1 *' provided)
-fentry:func_1 { $ret = socket_cookie(args.foo1); }
-                       ~~~~~~~~~~~~~~~~~~~~~~~~
-)" });
-  test("kprobe:tcp_shutdown { $ret = socket_cookie((struct sock *)arg0); }",
-       Error{ R"(
-stdin:1:30-64: ERROR: socket_cookie can not be used with "kprobe" probes
```

Comment:
I can't find where this is checked but seems like we should add this as a comptime check in the macro.